project/GenTupleAggregators.scala

```diff
@@ -66,8 +66,17 @@ object MapAggregator {
   }
 
   def genMapMethods(max: Int, isMonoid: Boolean = false): String = {
+    val aggType = if (isMonoid) "Monoid" else ""
+
+    // there's no Semigroup[Tuple1[T]], so just use T as intermediary type instead of Tuple1[T]
+    val aggregatorForOneItem = s"""
+       |def apply[A, B, C](aggDef: (String, ${aggType}Aggregator[A, B, C])): ${aggType}Aggregator[A, B, Map[String, C]] = {
+       |  val (key, agg) = aggDef
+       |  agg.andThenPresent(value => Map(key -> value))
+       |}
+    """.stripMargin
+
     (2 to max).map(i => {
```

Comment:
I guess that would cause needless boxing in the middle. Your original approach is better in that regard.